### ISSUE TITLE
Multivariate proposals for MH current not supported

### ISSUE THREAD

**Issue by torfjelde** [TuringLang/Turing.jl#906]
With the current implementation of `MH`, it's not possible to specify _multivariate_ proposals. 

MVE:

```julia
julia> using Turing

julia> # Model definition.
       @model demo() = begin
           x ~ MvNormal(zeros(2), ones(2))
       end
demo (generic function with 2 methods)

julia> m = demo()
Turing.Model{Tuple{:x},Tuple{},getfield(Main, Symbol("###inner_function#438#26")),NamedTuple{(),Tuple{}},NamedTuple{(),Tuple{}}}(getfield(Main, Symbol("###inner_function#438#26"))(Core.Box(getfield(Main, Symbol("###inner_function#438#26"))(#= circular reference @-2 =#))), NamedTuple(), NamedTuple())

julia> sample(m, MH(10, (:x, x -> MvNormal(x, 0.1))))
ERROR: MethodError: no method matching support(::MvNormal{Float64,PDMats.PDiagMat{Float64,Array{Float64,1}},Array{Float64,1}})
Closest candidates are:
  support(::DiscreteNonParametric) at /home/tor/.julia/packages/Distributions/0Wogo/src/univariate/discrete/discretenonparametric.jl:60
  support(::Union{Type{D<:Distribution{Univariate,Continuous}}, D<:Distribution{Univariate,Continuous}}) where D<:Distribution{Univariate,Continuous} at /home/tor/.julia/packages/Distributions/0Wogo/src/univariates.jl:130
  support(::Union{Type{D<:Distribution{Univariate,Discrete}}, D<:Distribution{Univariate,Discrete}}) where D<:Distribution{Univariate,Discrete} at /home/tor/.julia/packages/Distributions/0Wogo/src/univariates.jl:131
Stacktrace:
 [1] assume(::Turing.Sampler{MH{(:x,)}}, ::MvNormal{Float64,PDMats.PDiagMat{Float64,Array{Float64,1}},Array{Float64,1}}, ::Turing.Core.RandomVariables.VarName{:x}, ::Turing.Core.RandomVariables.VarInfo{NamedTuple{(:x,),Tuple{Turing.Core.RandomVariables.Metadata{Dict{Turing.Core.RandomVariables.VarName{:x},Int64},Array{MvNormal{Float64,PDMats.PDiagMat{Float64,Array{Float64,1}},Array{Float64,1}},1},Array{Turing.Core.RandomVariables.VarName{:x},1},Array{Float64,1},Array{Set{Turing.Selector},1}}}},Float64}) at /home/tor/.julia/dev/Turing/src/inference/mh.jl:197
 [2] macro expansion at /home/tor/.julia/dev/Turing/src/core/compiler.jl:101 [inlined]
 [3] macro expansion at ./REPL[6]:3 [inlined]
 [4] (::getfield(Main, Symbol("###inner_function#438#26")))(::Turing.Core.RandomVariables.VarInfo{NamedTuple{(:x,),Tuple{Turing.Core.RandomVariables.Metadata{Dict{Turing.Core.RandomVariables.VarName{:x},Int64},Array{MvNormal{Float64,PDMats.PDiagMat{Float64,Array{Float64,1}},Array{Float64,1}},1},Array{Turing.Core.RandomVariables.VarName{:x},1},Array{Float64,1},Array{Set{Turing.Selector},1}}}},Float64}, ::Turing.Sampler{MH{(:x,)}}, ::Turing.Model{Tuple{:x},Tuple{},getfield(Main, Symbol("###inner_function#438#26")),NamedTuple{(),Tuple{}},NamedTuple{(),Tuple{}}}) at /home/tor/.julia/dev/Turing/src/core/compiler.jl:438
 [5] #call#3 at /home/tor/.julia/dev/Turing/src/Turing.jl:73 [inlined]
 [6] Model at /home/tor/.julia/dev/Turing/src/Turing.jl:73 [inlined]
 [7] runmodel!(::Turing.Model{Tuple{:x},Tuple{},getfield(Main, Symbol("###inner_function#438#26")),NamedTuple{(),Tuple{}},NamedTuple{(),Tuple{}}}, ::Turing.Core.RandomVariables.VarInfo{NamedTuple{(:x,),Tuple{Turing.Core.RandomVariables.Metadata{Dict{Turing.Core.RandomVariables.VarName{:x},Int64},Array{MvNormal{Float64,PDMats.PDiagMat{Float64,Array{Float64,1}},Array{Float64,1}},1},Array{Turing.Core.RandomVariables.VarName{:x},1},Array{Float64,1},Array{Set{Turing.Selector},1}}}},Float64}, ::Turing.Sampler{MH{(:x,)}}) at /home/tor/.julia/dev/Turing/src/core/RandomVariables.jl:694
 [8] #sample#39(::Bool, ::Nothing, ::Int64, ::Function, ::Turing.Model{Tuple{:x},Tuple{},getfield(Main, Symbol("###inner_function#438#26")),NamedTuple{(),Tuple{}},NamedTuple{(),Tuple{}}}, ::MH{(:x,)}) at /home/tor/.julia/dev/Turing/src/inference/mh.jl:139
 [9] sample(::Turing.Model{Tuple{:x},Tuple{},getfield(Main, Symbol("###inner_function#438#26")),NamedTuple{(),Tuple{}},NamedTuple{(),Tuple{}}}, ::MH{(:x,)}) at /home/tor/.julia/dev/Turing/src/inference/mh.jl:113
 [10] top-level scope at none:0
```

From a quick look at the code, it seems like the following lines is the issue:

https://github.com/TuringLang/Turing.jl/blob/2b4ca5db05518040d90fddfbbc7912474e36e059/src/inference/mh.jl#L195-L200

E.g. `MvNormal` does not implement the `support` function. Even if it did, `<` does not work between arrays due to no ordering on R^n.

A quick (and imo, sufficient) fix is to assume the user know what they're doing and just drop the `support` check.

**Comment by cpfiffer** [comment]
Instead of dropping the support check, could we just have a `elseif typeof(distribution) == MvNormal` line added to the if block and treat it as a special case?

**Comment by torfjelde** [comment]
> Instead of dropping the support check, could we just have a elseif typeof(distribution) == MvNormal line added to the if block and treat it as a special case?

I think this is the case for _any_ `MultivariateDistribution`, not just `MvNormal`. It's also a problem if the user pass a custom distribution without a `support` defined.

**Comment by torfjelde** [comment]
Should this be closed? Thought this was fixed in #910

**Comment by cpfiffer** [comment]
I think we're fine to close this for now.
